### PR TITLE
Add `resolve_template` method to `EasyConfig` class

### PR DESCRIPTION
**Pull request by Flamefire**
(created using `eb --new-pr`)

Not only does this reduce the (non-trivial) code duplication, it also allows e.g. easyblocks to resolve a template without setting an EC parameter. e.g. `self.cfg.resolve_template('lib/%(pyshortver)s')` might be useful.

### CODE REVIEW

**Review by boegel — CHANGES_REQUESTED** [review]
@Flamefire Can you look into adding a dedicated test for the `resolve_template` method?

**Comment by Flamefire** [comment]
> @Flamefire Can you look into adding a dedicated test for the `resolve_template` method?

I though it was not necessary: The tests for checking the template values already cover this via the getters and a test if `ec.resolve_template(x) == resolve_template(x, ec.template_values)` doesn't seem to provide much value as that is basically the implementation of the method.

So only thing which I'd say is worth checking, is whether the template dict gets initialized first time this gets called to detect easier if it gets removed by some later change, although the other tests would fail already.   
But I'm not sure how to "legally" create an easyconfig instance without the template_values being initialized.

**Comment by Flamefire** [comment]
I added a test and tried to trigger the case where `template_values` is not set. Turns out this is impossible, so I removed the condition.

File: easybuild/framework/easyconfig/easyconfig.py [hunk 1]
```diff
@@ -1755,6 +1755,10 @@ def _generate_template_values(self, ignore=None):
             if self.template_values[key] is None:
                 del self.template_values[key]
 
+    def resolve_template(self, value):
+        """Resolve all templates in the given value using this easyconfig"""
+        return resolve_template(value, self.template_values)
```
Comment (boegel) [inline easybuild/framework/easyconfig/easyconfig.py:1760]:
```suggestion
        if not self.template_values:
            self.generate_template_values()
        return resolve_template(value, self.template_values)
```

**Review by boegel — CHANGES_REQUESTED** [review]

**Review by boegel — APPROVED** [review]
lgtm

**Pull request merged**